test/jdk/java/util/Currency/ValidateISO4217.java

```diff
@@ -113,10 +140,86 @@ public class ValidateISO4217 {
             {"XK", "EUR", "978", "2"},      // Kosovo
     };
     private static SimpleDateFormat format = null;
+    private static final String MODULE_PATCH_LOCATION =
+            System.getProperty("test.classes") + "/java/util/";
+    private static final String MOCKED_TIME = System.getProperty("MOCKED.TIME");
+
+    // Classes that should mock System.currentTimeMillis()
+    private static final String[] CLASSES =
+            Stream.concat(
+                Stream.of("Currency.class"),
+                Arrays.stream(Currency.class.getDeclaredClasses())
+                        .map(c -> "Currency$" + c.getSimpleName() + ".class")
+            ).toArray(String[]::new);
+
+    // "check" invocation only runs the main method (and not any tests) to determine if the
+    // future time checking is correct
+    public static void main(String[] args) {
```

Comment:
It would probably helpful to check if the patched Currency class does exist or not. Same for the `MOCKED.TIME=true` case.